### PR TITLE
store/tikv: forward requests by unary call

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

Support forwarding requests by unary call.

### What is changed and how it works?

What's Changed:

Add a `ForwardedHost` field to Request. When it is set, the gRPC `metadata{"tikv-forwarded-host": ForwardedHost}` will be attached to the request. TiKV which receives the request will check it and forward the request to the `ForwardedHost` as a proxy. It's useful when network partition occurs.

Because gRPC doesn't support setting the metadata for each request in the stream. We don't use BatchCommands for forwarding in this PR.

### Related changes
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test 
Has been tested with https://github.com/pingcap/tidb/pull/23244.

Side effects

### Release note 

- No release note.

### CODE REVIEW

File: store/tikv/client_test.go [hunk 1]
```diff
@@ -46,15 +49,10 @@ var _ = Suite(&testClientSuite{})
...
-	newConf.TiKVClient.MaxBatchSize = size
-	config.StoreGlobalConfig(&newConf)
-}
-
 func (s *testClientSerialSuite) TestConn(c *C) {
-	maxBatchSize := config.GetGlobalConfig().TiKVClient.MaxBatchSize
-	setMaxBatchSize(0)
+	defer config.UpdateGlobal(func(conf *config.Config) {
+		conf.TiKVClient.MaxBatchSize = 0
+	})()
```
Comment (MyonKeminta) [inline store/tikv/client_test.go:55]: How does this set the config to 0 now and restore it after the test?

**Review by MyonKeminta — APPROVED** [review]

File: store/tikv/config/config.go [hunk 2]
```diff
@@ -119,6 +119,18 @@ func StoreGlobalConfig(config *Config) {
 	globalConf.Store(config)
 }
 
+// UpdateGlobal updates the global config, and provide a restore function that can be used to restore to the original.
+func UpdateGlobal(f func(conf *Config)) func() {
```
[the remark this answers is not in the document]
Comment (youjiali1995) [reply store/tikv/config/config.go:123]: It will be extracted to a separated repo and there are many functions like this one in 'store/tikv'.

**Pull request merged**